src/System.Private.CoreLib/shared/System/Math.cs

```diff
@@ -26,16 +26,6 @@ public static partial class Math
 
         public const double PI = 3.14159265358979323846;
 
-        private const int maxRoundingDigits = 15;
```

Comment:
Why is this static constructor a problem? It is before-field-init, so it should not cost anything unless you actually access the fields.

Reply:
We assume in many places that before-field-init static constructors costs nothing unless you actually access the fields.

Reply:
Yeah, it's probably better to update our tooling to catch up with that.